clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -730,7 +730,13 @@ getKernelInvocationKind(FunctionDecl *KernelCallerFunc) {
 }
 
 static CXXRecordDecl *getKernelObjectType(FunctionDecl *Caller) {
```

Comment:
```suggestion
static const CXXRecordDecl *getKernelObjectType(FunctionDecl *Caller) {
```

Reply:
If we can, this whole function should just return a const-type.  I believe we can, it is more const correct, and it saves us a few casts.

Reply:
I tried that initially, but the const escapes to affect a lot of code, including some outside SemaSYCL (because of the visitor, maybe?).  This approach is the least intrusive right now.

Reply:
I actually just tried it, and other than this function, only 2 other lines are changed!  I think it is definitely worth doing here.

2 calls in SyclKernelBodyCreator need to change (Ctor and CreateKernelObjClone), plus the variable in ConstructOpenCLKernel.

I'd say it is definitely worth doing as const-casts are typically risky.

Reply:
^^ I was on an older version of the repo apparently.  I did a pull and it is a bit more involved.  If it is not non-const for a good reason, I'll likely submit a patch to do so.

Reply:
Thanks!  I rebased and got rid of that const cast.